bindings/python/src/transform.rs

```diff
@@ -0,0 +1,142 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::{error, fmt, sync::Arc};
+use iceberg::spec::Transform;
+use iceberg::transform::create_transform_function;
+use iceberg::Error;
+
+use arrow::{
+    array::{make_array, Array, ArrayData, ArrayRef},
+    error::ArrowError,
+    ffi::{from_ffi, to_ffi},
+};
+use libc::uintptr_t;
+use pyo3::{exceptions::PyOSError, exceptions::PyValueError, prelude::*};
+
+#[derive(Debug)]
+enum PyO3ArrowError {
```

Comment:
I believe most ffi related to arrow has been handled in [`arrow-rs/pyarrow`](https://github.com/apache/arrow-rs/blob/master/arrow/src/pyarrow.rs). Would you like to take a look? Maybe we can reuse them.